src/cuda/cuda-shader-object.cpp

```
@@ -60,6 +60,18 @@ void shaderObjectSetBinding(
         memcpy(dst + offset.uniformOffset, &handle, sizeof(handle));
         break;
     }
+    case slang::BindingType::CombinedTextureSampler:
+    {
+        TextureViewImpl* textureView = checked_cast<TextureViewImpl*>(slot.resource.get());
+        SamplerImpl* sampler = checked_cast<SamplerImpl*>(slot.resource2.get());
+        uint64_t handle = 0;
+        if (textureView && sampler)
+        {
+            handle = textureView->getTexObjectWithSamplerSettings(sampler->m_samplerSettings);
+        }
```

Shouldn't we report an error if either `textureView` or `sampler` is missing, rather than just silently not bind it?

---

This is in line with other backends. We generally want to catch errors like that in the validation layer and not pay the cost when not needed.

---

Good point